pandas_gbq/tests/test_gbq.py

```diff
@@ -402,129 +371,61 @@ def test_read_gbq_with_corrupted_private_key_json_should_fail(self):
                 private_key=re.sub('[a-z]', '9', _get_private_key_contents()))
 
 
-class TestReadGBQIntegration(object):
-
-    @classmethod
-    def setup_class(cls):
-        # - GLOBAL CLASS FIXTURES -
-        #   put here any instruction you want to execute only *ONCE* *BEFORE*
-        #   executing *ALL* tests described below.
-
-        _skip_if_no_project_id()
-
-        _setup_common()
-
-    def setup_method(self, method):
-        # - PER-TEST FIXTURES -
-        # put here any instruction you want to be run *BEFORE* *EVERY* test is
-        # executed.
-        pass
-
-    @classmethod
-    def teardown_class(cls):
-        # - GLOBAL CLASS FIXTURES -
-        # put here any instruction you want to execute only *ONCE* *AFTER*
-        # executing all tests.
-        pass
-
-    def teardown_method(self, method):
-        # - PER-TEST FIXTURES -
-        # put here any instructions you want to be run *AFTER* *EVERY* test is
-        # executed.
-        pass
-
-    def test_should_read_as_user_account(self):
-        _skip_local_auth_if_in_travis_env()
+def test_should_read(project, credentials):
 
-        query = 'SELECT "PI" AS valid_string'
-        df = gbq.read_gbq(query, project_id=_get_project_id())
-        tm.assert_frame_equal(df, DataFrame({'valid_string': ['PI']}))
+    query = 'SELECT "PI" AS valid_string'
+    df = gbq.read_gbq(query, project_id=project, private_key=credentials)
+    tm.assert_frame_equal(df, DataFrame({'valid_string': ['PI']}))
 
-    def test_should_read_as_service_account_with_key_path(self):
-        _skip_if_no_private_key_path()
-        query = 'SELECT "PI" AS valid_string'
-        df = gbq.read_gbq(query, project_id=_get_project_id(),
-                          private_key=_get_private_key_path())
-        tm.assert_frame_equal(df, DataFrame({'valid_string': ['PI']}))
-
-    def test_should_read_as_service_account_with_key_contents(self):
-        _skip_if_no_private_key_contents()
-        query = 'SELECT "PI" AS valid_string'
-        df = gbq.read_gbq(query, project_id=_get_project_id(),
-                          private_key=_get_private_key_contents())
-        tm.assert_frame_equal(df, DataFrame({'valid_string': ['PI']}))
-
-
-class TestReadGBQIntegrationWithServiceAccountKeyPath(object):
-
-    @classmethod
-    def setup_class(cls):
-        # - GLOBAL CLASS FIXTURES -
-        #   put here any instruction you want to execute only *ONCE* *BEFORE*
-        #   executing *ALL* tests described below.
-
-        _skip_if_no_project_id()
-        _skip_if_no_private_key_path()
 
-        _setup_common()
+class TestReadGBQIntegration(object):
 
-    def setup_method(self, method):
+    @pytest.fixture(autouse=True)
+    def setup(self, project, credentials):
```

Comment:
Since queries can be pretty long-running, I don't think we really want to use all 3+ credentials types here.